### PR TITLE
Make function dependencies explicit

### PR DESCRIPTION
**Pull request by vicb**
We (opennext-cloudflare) plan to reuse code from opennext-aws.

For that we would like to extract code out of `build.ts` so that we can call those functions from our own `build.ts`.

As a first step, this PR explicit function dependencies.

The PR contains multiple commits to ease the review.

### CODE REVIEW

File: packages/open-next/src/build/helper.ts [hunk 1]
```diff
@@ -17,37 +17,69 @@ const __dirname = url.fileURLToPath(new URL(".", import.meta.url));
...
+        logger.info("Monorepo detected at", currentPath);
+      }
+      return { root: currentPath, packager: found.packager };
+    }
+    currentPath = path.dirname(currentPath);
+  }
+
+  // note: a lock file (package-lock.json, yarn.lock, or pnpm-lock.yaml) is
+  //       not found in the app's directory or any of its parent directories.
+  //       We are going to assume that the app is not part of a monorepo.
```
Comment (conico974) [inline packages/open-next/src/build/helper.ts:79]: We may want to add at least a warning here. I don't think this is supposed to happen

File: packages/open-next/src/build.ts [hunk 2]
```diff
@@ -198,47 +167,48 @@ function printHeader(header: string) {
...
 
-function printOpenNextVersion() {
-  const { openNextVersion } = options;
-  logger.info(`OpenNext v${openNextVersion}`);
+function printOpenNextVersion(options: BuildOptions) {
+  logger.info(`OpenNext v${options.openNextVersion}`);
 }
 
-function initOutputDir(tempDir: string) {
+function initOutputDir(srcTempDir: string, options: BuildOptions) {
```
Comment (conico974) [inline packages/open-next/src/build.ts:178]: I guess we could move `srcTempDir` inside options as well

**Review by conico974 — APPROVED** [review]
Just 2 little comment, but other than that, it looks good to me

File: packages/open-next/src/build/helper.ts (hunk 1 above)
Comment (vicb) [reply packages/open-next/src/build/helper.ts:79]: I'll address in a follow up PR

File: packages/open-next/src/build.ts (hunk 2 above)
Comment (vicb) [reply packages/open-next/src/build.ts:178]: I'll address in a follow up PR

**Pull request merged**